Simple Incremental/Assets/Scripts/ProjectileWeapon.cs

```diff
@@ -0,0 +1,52 @@
+﻿using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+
+public class ProjectileWeapon : MonoBehaviour
+{
+    [SerializeField]
+    GameObject projectilePrefab = null;
+    [SerializeField]
+    string fireButtonName = "Fire1";
+
+    public SpriteRenderer spriteRenderer = null;
+    public Sprite projectileSprite = null;
+    public float projectileSpeed = 1f;
+    public int maxPenetrations = 1;
+    public float falloffTime = 1f;
+    public int damage = 1;
+    public float reloadSpeed = 1f;
+    Camera mainCam = null;
+    Queue<Projectile> projectiles = null;
+
+    public void Awake()
+    {
+        spriteRenderer = GetComponent<SpriteRenderer>();
+        mainCam = Camera.main;
+    }
+
+    private void Start()
+    {
+        projectiles = ProjectileManager.instance.projectiles;
+    }
+
+    void Update()
+    {
+        if (Input.GetButtonDown(fireButtonName))
```

Comment:
Seems odd to have inputs int your Projectile weapon code. Would it be better to have a input handler script that ends up triggering or calling other functions

Reply:
There is no input handler.

This is acting as a "WeaponController". I can segment apart the input logic into a separate controller class, and the data and trigger method stays here.